blip-tap.md

```diff
@@ -919,149 +919,251 @@ order to clamp down on price volatility exposure.
-p2p message `tap_rfq` is sent with the following structure:
-
-1. type: ?? (`tap_rfq`)
-2. data:
-     * [`32*byte`:`rfq_id`]
-     * [`32*byte`:`asset_id`]
-     * [`BigSize`:`asset_amt`]
-     * [`BigSize`:`suggested_rate_tick`]
+An RFQ negotiation is initiated when a requesting peer sends a new quote request
+p2p message called `tap_rfq` to their channel counterparty.
+
+In this message, the peer includes the asset they wish to obtain (identified by
+`in_asset_id`/`in_asset_group_key`) and the asset they wish to divest
+(identified by `out_asset_id`/`out_asset_group_key`).
+
+The `tap_rfq` message has a BOLT 01 wire message type integer of 52884, falling
+within the custom message range. Its TLV stream consists of the following TLV
+fields:
+
+| TLV Type | Field Name                | Field Type |
+|----------|---------------------------|------------|
+| 0        | version                   | 1*byte     |
+| 1        | rfq_id                    | 32*byte    |
+| 2        | transfer_type             | 1*byte     |
+| 3        | expiry                    | 8*byte     |
+| 4        | in_asset_id               | 32*byte    |
+| 5        | in_asset_group_key        | 33*byte    |
+| 6        | out_asset_id              | 32*byte    |
+| 7        | out_asset_group_key       | 33*byte    |
+| 8        | in_asset_to_btc           | FixedPoint |
+| 9        | out_asset_to_btc          | FixedPoint |
+| 10       | max_transferable_in_asset | BigSize    |
 
 where:
 
-* `rfq_id` is a randomly generate 32-byte value to uniquely identify this RFQ
-  request
-* `asset_id` is the asset ID of the asset the receiver intends to receive
-* `asset_amt` is the amount of units of said asset
-* `suggested_rate_tick` is the internal unit used for asset conversions. A tick
-  is 1/10000th of a currency unit. It gives us up to 4 decimal places of
-  precision (0.0001 or 0.01% or 1 bps). As an example, if the BTC/USD rate was
-  $61,234.95, then we multiply that by 10,000 to arrive at the `usd_rate_tick`:
-  `$61,234.95 * 10000 = 612,349,500`.  To convert back to our normal rate, we
-  decide by `10,000` to arrive back at `$61,234.95`.
-
-Given valid `rfq_id`, we then define an `tap_rfq_scid` by taking the last `8`
-bytes of the `rfq_id` and interpreting them as a 64-bit integer. 
+* `version` is the version of the quote request message; the current version
+  is `0`.
+* `rfq_id` is a randomly generated 32-byte value to uniquely identify this RFQ
+  request.
+* `transfer_type` (`uint8`): Specifies the type of transaction that will take
+  place if the quote request results in an accepted agreement. It is set to:
+    * `0` when the transfer type is unspecified.
+    * `1` when the requesting peer is attempting to pay a lightning invoice.
+    * `2` when the requesting peer is seeking to receive funds corresponding to
+      a lightning invoice.
+* `expiry` is the Unix timestamp in seconds representing the exact time after
```

Comment:
Do we need to mention endianness? Could just mention somewhere that we encode all numbers that aren't `BigSize` as big endian.